mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMailboxMapper.java

```diff
@@ -187,14 +187,16 @@ public boolean apply(MailboxAnnotation input) {
     }
 
     @Override
-    public synchronized void setAnnotation(Mailbox mailbox, List<MailboxAnnotation> mailboxAnnotations) throws MailboxException {
+    public void updateAnnotations(Mailbox mailbox, List<MailboxAnnotation> mailboxAnnotations) throws MailboxException {
         InMemoryId mailboxId = (InMemoryId) mailbox.getMailboxId();
 
         for (MailboxAnnotation mailboxAnnotation : mailboxAnnotations) {
-            if (mailboxAnnotation.getValue().isPresent()) {
-                mailboxesAnnotations.put(mailboxId, mailboxAnnotation.getKey(), mailboxAnnotation.getValue().get());
-            } else {
-                mailboxesAnnotations.remove(mailboxId, mailboxAnnotation.getKey());
+            synchronized(mailboxAnnotations) {
```

Comment:
Here I can observe what looks like a "trasaction" in some intermediary step. I would prefer to see this lock around the for (better consistency)